File: 36 - Computing Statistics/Python/main.py
from math import sqrt


def generate_list_ints(output):
    l = list()
    while True:
        inp = input(output)
        if inp != "":
            try:
                l.append(int(inp))
            except ValueError:
                print("NaN")
        else:
            return l


def mean(nums):
    return sum(nums) / len(nums)


def standard_deviation(nums):
    meann = mean(nums)
    nums = [((i - meann) ** 2) for i in nums]
    return sqrt(sum(nums) / (len(nums) - 1))


def main():
    times = generate_list_ints("Enter a number: ")
    print(f"Average: {mean(times)}")
    print(f"Maximum: {max(times)}")
    print(f"Minimum: {min(times)}")
    print(f"Standard deviation: {standard_deviation(times)}")


if __name__ == "__main__":
    main()

File: 36 - Computing Statistics/Golang/main.go
package main

import (
	"bufio"
	"fmt"
	"math"
	"os"
	"strconv"
)

func generateListInts(output string) []int {
	list := make([]int, 0)
	scanner := bufio.NewScanner(os.Stdin)
	for {
		fmt.Print(output)
		scanner.Scan()
		inp := scanner.Text()
		if inp == "" {
			return list
		}
		num, err := strconv.Atoi(inp)
		if err != nil {
			fmt.Println(err)
		} else {
			list = append(list, num)
		}
	}
}

func standardDeviation(nums []int) float64 {
	mean := mean(nums)
	newnums := make([]float64, 0, 0)
	for _, val := range nums {
		fval := float64(val)
		newnums = append(newnums, ((fval - mean) * (fval - mean)))
	}
	return math.Sqrt(sumFloats(newnums) / float64(len(newnums)-1))
}

func mean(nums []int) float64 {
	return float64(sumInts(nums)) / float64(len(nums))
}

func main() {
	numbers := generateListInts("Enter a number: ")
	fmt.Println("Numbers: ", numbers)
	fmt.Printf("Average: %f\n", mean(numbers))
	fmt.Printf("Minimum: %d\n", minInt(numbers))
	fmt.Printf("Maximum: %d\n", maxInt(numbers))
	fmt.Printf("Standard deviation: %f\n", standardDeviation(numbers))
}
